components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/oauth2_key_validation_provider.bal

```diff
@@ -219,6 +219,24 @@ public type OAuth2KeyValidationProvider object {
 
     }
 
+    public function setScopesToPrincipalComponent(xml responseXml) {
+            string[] scopes = [];
+            int i = 0;
+            // Iterate through scopes.
+            foreach xml|string scope in responseXml/<apim:scopes>/* {
+                if (scope is xml) {
```

Comment:
If scope is string then what are we returning

Reply:
Added an error message if scope is string